data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaProducerProperties.java

```diff
@@ -0,0 +1,250 @@
+
+    @JsonProperty("buffer_memory")
+    private String bufferMemory = DEFAULT_BYTE_CAPACITY;
+
+    @JsonProperty("compression_type")
+    private String compressionType;
+
+    @JsonProperty("retries")
+    private int retries;
+
+    @JsonProperty("batch_size")
+    private int batchSize;
+
+    @JsonProperty("client_dns_lookup")
+    private String clientDnsLookup;
+
+    @JsonProperty("client_id")
+    private String clientId;
+
+    @JsonProperty("connections_max_idle")
+    private Duration connectionsMaxIdleMs;
+
+    @JsonProperty("delivery_timeout")
+    private Duration deliveryTimeoutMs;
+
+    @JsonProperty("linger_ms")
+    private Long lingerMs;
+
+    @JsonProperty("max_block")
+    private Duration maxBlockMs;
+
+    @JsonProperty("max_request_size")
+    private int maxRequestSize;
+
+    @JsonProperty("partitioner_class")
+    private Class partitionerClass;
+
+    @JsonProperty("partitioner_ignore_keys")
+    private Boolean partitionerIgnoreKeys;
+
+    @JsonProperty("receive_buffer")
+    private String receiveBufferBytes=DEFAULT_BYTE_CAPACITY;
+
+    @JsonProperty("request_timeout")
+    private Duration requestTimeoutMs;
+
+    @JsonProperty("send_buffer")
+    private String sendBufferBytes=DEFAULT_BYTE_CAPACITY;
+
+    @JsonProperty("socket_connection_setup_timeout_max")
+    private Duration socketConnectionSetupMaxTimeout;
+
+    @JsonProperty("socket_connection_setup_timeout")
+    private Duration socketConnectionSetupTimeout;
+
+    @JsonProperty("acks")
+    private String acks;
+
+    @JsonProperty("enable_idempotence")
+    private Boolean enableIdempotence;
+
+    @JsonProperty("interceptor_classes")
+    private List interceptorClasses;
+
+    @JsonProperty("max_in_flight_requests_per_connection")
+    private int maxInFlightRequestsPerConnection;
+
+    @JsonProperty("metadata_max_age")
+    private Duration metadataMaxAgeMs;
+
+    @JsonProperty("metadata_max_idle")
+    private Duration metadataMaxIdleMs;
+
+    @JsonProperty("metric_reporters")
+    private List metricReporters;
+
+    @JsonProperty("metrics_num_samples")
+    private int metricsNumSamples;
+
+    @JsonProperty("metrics_recording_level")
+    private String metricsRecordingLevel;
+
+    @JsonProperty("metrics_sample_window")
+    private Duration metricsSampleWindowMs;
+
+    @JsonProperty("partitioner_adaptive_partitioning_enable")
+    private boolean partitionerAdaptivePartitioningEnable;
+
+    @JsonProperty("partitioner_availability_timeout")
+    private Duration partitionerAvailabilityTimeoutMs;
+
+    @JsonProperty("reconnect_backoff_max")
+    private Duration reconnectBackoffMaxMs;
+
+    @JsonProperty("reconnect_backoff")
+    private Duration reconnectBackoffMs;
+
+    @JsonProperty("retry_backoff")
+    private Duration retryBackoffMs;
+
+
+    public String getCompressionType() {
+        return compressionType;
+    }
+
+    public int getRetries() {
+        if (retries == 0) {
```

Comment:
might be better to assign default value?

Reply:
kafka already has default values.